runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/cache/KafkaCacheCursorFactory.java

```diff
@@ -1121,7 +1123,11 @@ private KafkaFilterCondition asKeyCondition(
         long mask,
         KafkaKeyFW key)
     {
-        return new KafkaFilterCondition.Key(mask, checksum, key);
+        final OctetsFW value = key.value();
+
+        return value == null ?
+            new KafkaFilterCondition.Key(mask, checksum, nullKeyRO) : new KafkaFilterCondition.Key(mask, checksum, key);
+
```

Comment:
This is closer, but we still make the copy of `nullKeyRO` each time we construct a new `KafkaFilterCondition.Key`.

Follow the constructor hierarchy up through `super` to see the copy in `Equals`.